### PR TITLE
Squash multiple consecutive spaces

### PR DESCRIPTION
**Pull request by jwang47**

### CODE REVIEW

File: common-cli/src/main/java/co/cask/common/cli/util/Parser.java [hunk 1]
```diff
@@ -49,12 +49,18 @@
...
+    char[] chars = input.toCharArray();
+    for (int i = 0; i < chars.length; i++) {
+      char ch = chars[i];
       switch (state) {
         case EMPTY:
           if (ch == SEPARATOR) {
             splitInput.add(builder.toString());
             builder.setLength(0);
+            // skip consecutive SEPARATORs
+            while (chars[i+1] == SEPARATOR && i+1 < chars.length) {
```
Comment (albertshau) [inline common-cli/src/main/java/co/cask/common/cli/util/Parser.java]: have the `i+1 < chars.length` first, otherwise you can get an array out of bounds

Comment (albertshau) [inline common-cli/src/main/java/co/cask/common/cli/util/Parser.java]: would be good to have a test case for this too

**Comment by albertshau** [comment]
one comment, then lgtm

File: common-cli/src/main/java/co/cask/common/cli/util/Parser.java (hunk 1 above)
Comment (anwar6953) [inline common-cli/src/main/java/co/cask/common/cli/util/Parser.java]: The edge case seems improper.
Having the `i + 1 < chars.length` check as the first condition is one way to fix it.

Comment (jwang47) [inline common-cli/src/main/java/co/cask/common/cli/util/Parser.java]: True, will fix

**Pull request merged**